### ISSUE TITLE
System.Windows.Forms assembly (WinForms) can no longer be loaded with Add-Type -AssemblyName

### ISSUE THREAD

**Issue by mklement0** [PowerShell/PowerShell#18851]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

Related: the ability to load the WinForms assembly via  `using assembly System.Windows.Forms` has been broken since using  before v7..0:
 * see https://github.com/PowerShell/PowerShell/issues/11856

Now, even `Add-Type -Assembly System.Window.Forms` doesn't work anymore:

```powershell
Add-Type -AssemblyName System.Windows.Forms; [System.Windows.Forms.Form].FullName
```

### Expected behavior

```console
System.Windows.Forms.Form
```

That is, the assembly should load, and the `[System.Windows.Forms.Form]` type should be available.

### Actual behavior

```console
Add-Type: Cannot find path 'C:\path\to\current\location\System.Windows.Forms.dll' because it does not exist.
InvalidOperation: Unable to find type [System.Windows.Forms.Form].
```

That is, the assembly is seemingly only looked for in the _current_ location, where it cannot be found.

Using an explicit GAC path to load the assembly does work, however:

```powershell
# OK, with full GAC path.
Add-Type -LiteralPath C:\Windows\Microsoft.NET\Framework64\v4.0.30319\System.Windows.Forms.dll; [System.Windows.Forms.Form]
```

### Environment data

```powershell
PowerShell Core v7.3.1 (.NET 7.0.0)
PowerShell Core v7.4.0-preview.1 (.NET 7.0.1) 
on Microsoft Windows 11 Pro (ARM 64-bit Processor; Version 22H2, OS Build: 22621.963)
```

**Comment by 237dmitry** [comment]
Assembly loaded, 7.3.1

**Comment by mklement0** [comment]
Mysterious, @237dmitry.

I've updated the initial post with environment details (Windows 11 22H2). Are you on a different version?
Does it still work for you if you run in a pristine session launched with `-NoProfile`?

I can only get the assembly to load if I use a full path into the GAC.
Once loaded that way, executing `Add-Type -Assembly System.Window.Forms` works (quiet no-op).

**Comment by 237dmitry** [comment]
With `-NoProfile`:

![Screenshot 2022-12-27 214434](https://user-images.githubusercontent.com/78153320/209710228-fad037e2-6475-47cc-8e72-068fa7d8129f.png)

Win 11 22H2.

**Comment by MartinGC94** [comment]
Can confirm that it also works for me on Windows 10:
![image](https://user-images.githubusercontent.com/42123497/209711521-7cf6a5e3-64cf-4680-99d0-6647a4a49f85.png)

**Comment by mklement0** [comment]
Mysterious; thanks for checking.

* Does 

```powershell
(Add-Type -AssemblyName System.Windows.Forms -PassThru | Select -First 1 | % Assembly).Location
```

return a path in the GAC for you? 

* Is that where the assembly is expected to be loaded from in PS Core too?

* What could get in the way of performing a GAC lookup in this case? (It works from WinPS on the same machine).

**Comment by MartinGC94** [comment]
Core returns: `C:\Program Files\PowerShell\7-preview\System.Windows.Forms.dll` Windows PowerShell returns: `C:\Windows\Microsoft.Net\assembly\GAC_MSIL\System.Windows.Forms\v4.0_4.0.0.0__b77a5c561934e089\System.Windows.Forms.dll`

**Comment by mklement0** [comment]
Thanks, @MartinGC94.

I think I know what the problem is:

Whether the WinForms and WPF assemblies ship with PowerShell Core depends on the installation method:
* YES:
    * MSI package
* NO:
    * `winget install Microsoft.PowerShell`
    * Installation via https://aka.ms/install-powershell.ps1 

Do you happen to know if this a known / by-design / documented discrepancy?

**Comment by mklement0** [comment]
Closed in favor of #18858
